### DISCUSSION TITLE
Candlestick and Corsshair x_offset

### DISCUSSION THREAD

**Discussion by forna** [highfestiva/finplot#202] [Ideas]
Congrats for the library, it is amazing!

I have customized the look quite a lot to make it similar to the IB TWS charts.
I think it is more correct to have the candlesticks on the right of the x grid (not in the middle), so in the class `CandlestickItem(FinPlotItem)` on line 1026 I have set the attribute `self.x_offset = 0.5` (instead of 0) directly in the library.
I would suggest to make this a customizable since it may be useful for others and so I don't have to update the library every time there is a new version ;)
I am trying to offset the Crosshair too, so it would again snap in the middle of the candlesticks.  
Any suggestions on how to achieve that?
Thank you!

**Before the change:**
![image](https://user-images.githubusercontent.com/1431006/120883831-c1848b00-c5df-11eb-9b7d-4def13ae1f47.png)

**After the change:**
![image](https://user-images.githubusercontent.com/1431006/120883545-13c4ac80-c5de-11eb-8080-5614a9eaafcd.png)

**Comment by forna** [comment]
Ok, I have found the solution for the candlesticks offset:
```
p = fplt.candlestick_ochl(self.widget.df[['date', 'open', 'close', 'high', 'low']])
p.x_offset = 0.5
```
If you have a hint on how to offset the crosshair it would be appreciated. 
Thanks

**Reply by highfestiva** [reply]
All you're doing is moving everything slightly to the right. I don't know what you're trying to achive, but probably not what you're getting. :)  If you want to show different times on the axis/crosshair, you do that by modifying the times you enter. If you want to display in a different time zone (other than your local), see this: https://github.com/highfestiva/finplot/wiki/Snippets#time-zone

**Comment by forna** [comment]
Well, this is how charts work for many trading applications.
This below is the default chart in Interactive Brokers TWS.
The gridlines and the initial times are on the left of the candlesticks, not in the middle.
Anyway not an issue, I will find some way to make the crosshair behave as I would like.

![image](https://user-images.githubusercontent.com/1431006/120991321-cbde8a80-c781-11eb-950c-bfc2d5536815.png)

![image](https://user-images.githubusercontent.com/1431006/120991339-d00aa800-c781-11eb-898d-b74423a17226.png)

**Reply by highfestiva** [reply]
Interesting! Tradingview, which I modelled from, does not do it that way. You probably want to override `fplt._clamp_xy()`, call the previous implementation, and add +0.5 to the X-value before returning to caller.
